src/main/java/hudson/plugins/git/util/Build.java

```diff
@@ -92,6 +92,21 @@ public Result getBuildResult() {
     }
 
     @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof Build)) {
+            return false;
+        } else {
+            Build otherBuild = (Build) o;
+            if (otherBuild.hudsonBuildNumber == this.hudsonBuildNumber
+                    && otherBuild.revision.equals(this.revision)) {
+                return true;
```

Comment:
I wasn't sure whether to include that or not - `revision` is used in `toString` while `marked` isn't, so I skipped `marked` here. But I'd be happy to add it.